cdap-security/src/main/java/co/cask/cdap/security/store/FileSecureStore.java

```diff
@@ -95,28 +104,34 @@
   }
 
   /**
-   * Stores an element in the secure store.
+   * Stores an element in the secure store. Although JCEKS supports overwriting keys the interface currently does not
+   * support it. If the key already exists then this method throws an AlreadyExistsException.
    * @param namespace The namespace this key belongs to.
    * @param name Name of the element to store.
    * @param data The data that needs to be securely stored.
    * @param description User provided description of the entry.
-   * @param properties Metadata associated with the data
-   * @throws IOException If it failed to store the key in the store.
+   * @param properties Metadata associated with the data.
+   * @throws NamespaceNotFoundException If the specified namespace does not exist.
+   * @throws AlreadyExistsException If the key already exists in the namespace. Updating is not supported.
+   * @throws IOException If there was a problem storing the key to the in memory keystore
+   * or if there was problem persisting the keystore.
    */
   @Override
   public void putSecureData(String namespace, String name, byte[] data, String description,
-                            Map<String, String> properties) throws IOException {
+                            Map<String, String> properties) throws Exception {
+    checkNamespaceExists(namespace);
     String keyName = getKeyName(namespace, name);
     SecureStoreMetadata meta = SecureStoreMetadata.of(name, description, properties);
     SecureStoreData secureStoreData = new SecureStoreData(meta, data);
     writeLock.lock();
     try {
       if (keyStore.containsAlias(keyName)) {
```

Comment:
so right now, the behavior will be different between File and KMS right? File will throw 409, KMS will throw 500 in this case?